Sources/Apodini/Semantic Model Builder/Model/Tree.swift

```diff
@@ -134,3 +134,46 @@ extension Node {
         return Node<U>(value: value, children: children)
     }
 }
+
+extension Node: CustomStringConvertible where T: CustomStringConvertible {
+    private var lines: [Substring] {
+        let children = self.children
+            .map { child in
+                child.lines.enumerated().map { (index, substring) -> Substring in
+                    let prefix = index == 0 ? "→ " : "  "
+                    return Substring(prefix + substring)
+                }
+            }
+            .flatMap { $0 }
+        
+        return value.description
+            .split(separator: "\n")
+            + children
+    }
+    
+    var description: String {
+        lines.joined(separator: "\n")
+    }
+}
+
+extension Node {
+    func collect() -> Set<T> where T: Hashable {
+        reduce(Set()) { (partialResults, next) in
+            var set = Set(arrayLiteral: next)
+            for result in partialResults {
+                set.formUnion(result)
+            }
+            return set
+        }
+    }
+    
+    func collect2<U>() -> Set<U> where T == Array<U>, U: Hashable {
```

Comment:
Can we maybe think of a better name than appending `2` as a suffix 🙃

Reply:
Yes, I renamed them to `Tree.collectValues`. The `2` would've indicated that we are dealing with a `Collection` inside the Tree "Collection".